src/reply/send_reply.rs

```diff
@@ -138,19 +118,7 @@ async fn _send_prefix_reply<'a, U, E>(
 
     Ok(Box::new(if let Some(mut response) = existing_response {
         response
-            .edit(ctx.serenity_context, |f| {
-                // Reset the message. We don't want leftovers of the previous message (e.g. user
-                // sends a message with `.content("abc")` in a track_edits command, and the edited
-                // message happens to contain embeds, we don't want to keep those embeds)
-                // (*f = Default::default() won't do)
-                f.content("");
-                f.set_embeds(Vec::new());
-                f.components(|b| b);
-                f.0.insert("attachments", serenity::json::json! { [] });
-
-                reply.to_prefix_edit(f);
-                f
-            })
+            .edit(ctx.serenity_context, builder.to_prefix_edit())
```

Comment:
I see you sneakily moved the reset code from here into to_prefix_edit. This subtly changes the behavior of to_prefix_edit-serialized CreateReply instances: fields that are not set will not be left alone, but reset to default. I suppose that's fine, because semantically we're treating CreateReply as the specification of a stand-alone message, not an amend to a potentially-existent message.

Reply:
I moved it as to_prefix_edit is now creating the builder, so there was no other choice.

Reply:
Huh, no. `to_prefix_edit` could still leave fields unset (i.e. `None`), which would leave those aspects of the message unchanged when sending the builder to Discord.